### PR TITLE
test: A DRAFT for EnforceSync

### PR DESCRIPTION
**Pull request by 17bit**
Signed-off-by: Xiaotian Lu <xtlu000@gmail.com>

@fgblomqvist @hsluoyz @nodece  @kibertoad

Hi~ I made a draft of EnforceSync API by generate function. This is the best way I can think of to handle Async and Sync.
Code is poor , this PR is only for review.

### CODE REVIEW

**Comment by hsluoyz** [comment]
@17bit plz fix this:

image

**Comment by fgblomqvist** [comment]
This is quite clever, but I'm not sure if it's the best approach 🤔

**Comment by nodece** [comment]
It looks great, but we should still check the value of generate.

File: src/coreEnforcer.ts [hunk 1]
```diff
@@ -395,8 +394,7 @@ export class CoreEnforcer {
 
       expression = this.getExpression(asyncCompile, expString);
       const context = { ...parameters, ...functions };
-      const result = asyncCompile ? await expression(context) : expression(context);
-
+      const result = asyncCompile ? yield expression(context) : expression(context);
```
Comment (nodece) [inline src/coreEnforcer.ts:397]: The result type is `{done: boolean, vaule: Promise<boolean> | boolean}`

Comment (17bit) [reply src/coreEnforcer.ts:397]: Thanks for your advice

**Pull request closed without merging**